cypress/integration/data-browser/graphs.spec.js

```diff
@@ -1,171 +1,154 @@
-    cy.get(".react-select__graph__control")
-      .click(0, 0, { force: true })
-      .get('[class*="-menu"]')
-      .find('[class*="-option"]')
-      .eq(0)
-      .click(0, 0, { force: true })
-      .find('[class*="-singleValue"]')
-      .contains("How has the number of applications changed?")
-    // Selects 2nd option from period selector start
-    cy.get(".react-select__period_start__control")
-      .click(0, 0, { force: true })
-      .get('[class*="-menu"]')
-      .find('[class*="-option"]')
-      .eq(1)
-      .contains("2018-Q2")
-      .click(0, 0, { force: true });
-    // Selects 6th option from period selector end
-    cy.get(".react-select__period_end__control")
-      .click(0, 0, { force: true })
-      .get('[class*="-menu"]')
-      .find('[class*="-option"]')
-      .eq(5)
-      .contains("2019-Q4")
-      .click(0, 0, { force: true });
-    // Update URL with period selection and series
-    let urlUpdate = buildURL(
-      baseURLToVisit,
-      "applications",
-      "2018-Q2",
-      "2019-Q4",
-      firstGraphEndpoint.series
-    );
-    cy.url("include", urlUpdate);
-    cy.url("eq", urlUpdate);
-    // Clicks `Show All Quarters` button
-    cy.get(".period-wrapper").get(".reset").click();
-    // Update URL with period selection and series
-    urlUpdate = buildURL(
-      baseURLToVisit,
-      "applications",
-      "2018-Q1",
-      "2021-Q4",
-      firstGraphEndpoint.series
-    );
-    cy.url("include", urlUpdate);
-    cy.url("eq", urlUpdate);
-  });
-
-  it("De-select and re-select a series and have the URL change/update the in the UI", () => {
-    let urlUpdate;
-    cy.visit(`${baseURLToVisit}/data-browser/graphs/quarterly`);
-    cy.get(".highcharts-color-0 > text").click(0, 0, { force: true });
-    urlUpdate = buildURL(baseURLToVisit, "applications", "2018-Q1", "2021-Q4", [
-      "Conventional Non-Conforming",
-      "FHA",
-      "HELOC",
-      "RHS/FSA",
-      "VA",
-    ]);
-    cy.url("include", urlUpdate);
-    cy.url("eq", urlUpdate);
-    cy.get(".highcharts-color-0 > text").click(0, 0, { force: true });
-    urlUpdate = buildURL(baseURLToVisit, "applications", "2018-Q1", "2021-Q4", [
-      "Conventional Conforming",
-      "Conventional Non-Conforming",
-      "FHA",
-      "HELOC",
-      "RHS/FSA",
-      "VA",
-    ]);
-    cy.url("include", urlUpdate);
-    cy.url("eq", urlUpdate);
-  });
+  it("Visits graph page and checks that the url contains correct query parameters", () => {
+    cy.visit(`${baseURLToVisit}/data-browser/graphs/quarterly`)
+    cy.wait(2000)
+    // Regex check graphName, periodLow, periodHigh and visibleSeries in the url
+    cy.url().then((url) => {
+      cy.url().should("match", /quarterly\/([a-z]{2,}-?)+/)
+      cy.url().should("match", /periodLow=20\d{2}-Q\d/)
+      cy.url().should("match", /periodHigh=20\d{2}-Q\d/)
+      cy.url().should(
+        "match",
+        /visibleSeries=([a-zA-Z-\/]+(%20)?[a-zA-Z-]*,?)+/
+      )
+    })
```

Comment:
We can remove the enclosing `cy.url().then((url) => {` since each parameter test does it's own `cy.url()` call.

Reply:
Updated test code.